nettests/nettests.go

```diff
@@ -273,6 +279,117 @@ func (c *Controller) Init(nt *mk.Nettest) error {
+			log.Debugf(
+				"%s: %s", color.RedString("failure.report_create"), err.Error(),
+			)
+		} else {
+			defer exp.CloseReport(ctx)
+			log.Debugf(color.RedString("status.report_create"))
+			reportID = sql.NullString{String: exp.ReportID(), Valid: true}
+		}
+	}
+
+	for idx, input := range inputs {
+		idx64 := int64(idx)
+		log.Debug(color.RedString("status.measurement_start"))
+		urlID := sql.NullInt64{Int64: 0, Valid: false}
+		if c.inputIdxMap != nil {
+			urlID = sql.NullInt64{Int64: c.inputIdxMap[idx64], Valid: true}
+		}
+		msmt, err := database.CreateMeasurement(
+			c.Ctx.DB, reportID, exp.TestName, resultID, c.msmtPath, urlID,
+		)
+		if err != nil {
+			return errors.Wrap(err, "failed to create measurement")
+		}
+		c.msmts[idx64] = msmt
+
+		measurement, err := exp.Measure(ctx, input)
+		if err != nil {
+			log.Debug(color.RedString("failure.measurement"))
+			if err := c.msmts[idx64].Failed(c.Ctx.DB, err.Error()); err != nil {
+				return errors.Wrap(err, "failed to mark measurement as failed")
+			}
+			continue
+		}
+
+		if c.Ctx.Config.Sharing.UploadResults {
+			// Implementation note: SubmitMeasurement will fail here if we did fail
+			// to open the report but we still want to continue. There will be a
+			// bit of a spew in the logs, perhaps, but stopping seems less efficient.
+			if err := exp.SubmitMeasurement(ctx, &measurement); err != nil {
+				log.Debug(color.RedString("failure.measurement_submission"))
+				if err := c.msmts[idx64].UploadFailed(c.Ctx.DB, err.Error()); err != nil {
+					return errors.Wrap(err, "failed to mark upload as failed")
+				}
+			} else if err := c.msmts[idx64].UploadSucceeded(c.Ctx.DB); err != nil {
+				return errors.Wrap(err, "failed to mark upload as succeeded")
+			}
+		}
+
+		if err := exp.SaveMeasurement(measurement, c.msmtPath); err != nil {
+			return errors.Wrap(err, "failed to save measurement on disk")
+		}
+		if err := c.msmts[idx64].Done(c.Ctx.DB); err != nil {
+			return errors.Wrap(err, "failed to mark measurement as done")
+		}
+
+		// We're not sure whether it's enough to log the error or we should
+		// instead also mark the measurement as failed. Strictly speaking this
+		// is an inconsistency between the code that generate the measurement
+		// and the code that process the measurement. We do have some data
+		// but we're not gonna have a summary. To be reconsidered.
+		tk, err := c.nt.GetTestKeys(measurement.TestKeys)
+		if err != nil {
+			log.WithError(err).Error("failed to obtain testKeys")
+			continue
+		}
+		log.Debugf("Fetching: %d %v", idx, c.msmts[idx64])
+		if err := database.AddTestKeys(c.Ctx.DB, c.msmts[idx64], tk); err != nil {
+			return errors.Wrap(err, "failed to add test keys to summary")
+		}
+	}
+
+	log.Debugf("status.end")
+	for idx, msmt := range c.msmts {
+		log.Debugf("adding msmt#%d to result", idx)
+		if err := msmt.AddToResult(c.Ctx.DB, c.res); err != nil {
+			return errors.Wrap(err, "failed to add to result")
+		}
+	}
+	return nil
+}
```

Comment:
I believe that some of the logic inside of this `Run` method is also present inside of `Init` which I see is no longer getting called. I suggest dropping the `Init` method as part of this PR as well.

Reply:
I cannot drop it until all tests have been converted to use `Run` rather than `Init`. Currently, I just converted the IM tests. If you're okay with this approach and feel like I succeeded in preserving the meaning of the original code in `Run`, then I'll be happy to follow up and make sure all the tests use `Run` rather than `Init`.

Reply:
I see. Yes I think this approach is valid and I would suggest we drop all the `Init` code and update the tests to use `Run` instead.

Reply:
Done!